docs/topics/arrays.md

```diff
@@ -31,32 +50,244 @@ fun main() {
+
+```Kotlin
+fun main() { 
+//sampleStart
+    val pairArray = arrayOf(Pair(1,1), Pair(2,2), Pair(3,3))
+
+    val unzipArray = pairArray.unzip()
+    println(unzipArray)
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="unzip-array-kotlin"}
+
+## Concatenate arrays
+
+Arrays of type `String` can be concatenated using the `+` operator.
+
+```kotlin
+fun main() {
+//sampleStart
+    val simpleArray = arrayOf("a", "b", "c")
+    val anotherArray = arrayOf("d", "e", "f")
+
+    println((simpleArray + anotherArray).joinToString())
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="concatenate-array-kotlin"}
+
+## Pass array as vararg
 
-Kotlin also has classes that represent arrays of primitive types without boxing overhead: `ByteArray`,
-`ShortArray`, `IntArray`, and so on. These classes have no inheritance relation to the `Array` class, but they
-have the same set of methods and properties. Each of them also has a corresponding factory function:
+To pass an array to the `vararg` parameter of a function, use the _spread_ operator (`*`).
 
 ```kotlin
-val x: IntArray = intArrayOf(1, 2, 3)
-x[0] = x[1] + x[2]
+val a = arrayOf(1, 2, 3)
+val list = asList(-1, 0, *a, 4)
 ```
 
+For more information, see [Variable number of args (varargs)](functions.md#variable-number-of-arguments-varargs).
+
+## Convert to collection
+
+To convert an array to a [collection](collections-overview.md) (`Set`, `List`, or `Map`), use the [`toSet()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/to-set.html), [`toList()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/to-list.html), [`toMap()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/to-map.html) functions respectively.
+
 ```kotlin
-// Array of int of size 5 with values [0, 0, 0, 0, 0]
-val arr = IntArray(5)
+fun main() {
+//sampleStart
+    val simpleArray = arrayOf("a", "b", "c", "c")
+    
+    // Convert to Set
+    println(simpleArray.toSet())
+    
+    // Convert to List
+    println(simpleArray.toList())
+
+    val pairArray = arrayOf(Pair(1,1), Pair(2,2), Pair(3,3))
+    
+    // Convert to Map
+    println(pairArray.toMap())
+    
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="convert-array-kotlin"}
 
-// Example of initializing the values in the array with a constant
-// Array of int of size 5 with values [42, 42, 42, 42, 42]
-val arr = IntArray(5) { 42 }
+## What's next?
 
-// Example of initializing the values in the array using a lambda
-// Array of int of size 5 with values [0, 1, 2, 3, 4] (values initialized to their index value)
-var arr = IntArray(5) { it * 1 } 
-```
\ No newline at end of file
+* Learn about other [basic types](basic-types.md).
+* Learn about [collections](collections-overview.md).
```

Comment:
Once again, big emphasis on the point that developers likely should've gone straight to the collections API anyway.